### DISCUSSION TITLE
How do I enable/disable the Interactive Rebase Editor?

### DISCUSSION THREAD

**Discussion by craph** [gitkraken/vscode-gitlens#1260] [Q&A] [answered]
Hi @eamodio ,

First, thank you for your beautiful work for GitLens :) 

Is there an easy way to turn back on the Interactive Rebase Editor if I have disabled it by mistake ?

Where is the settings to turn it back on ?

Thank you very much.

Best regards,

**Comment by eamodio** [comment, marked as answer]
@craph Thanks!

To enable the _Interactive Rebase Editor_ use the _GitLens: Enable Interactive Rebase Editor_ command from the command palette (`ctrl+shift+p` or `cmd+shift+p`).

To disable the _Interactive Rebase Editor_ use the _GitLens: Disable Interactive Rebase Editor_ command from the command palette (`ctrl+shift+p` or `cmd+shift+p`).

![image](https://user-images.githubusercontent.com/641685/102309373-9d03fc00-3f36-11eb-8c23-8c49453787b3.png)

Additionally, you can also open the GitLens interactive settings editor from the command palette (`ctrl+shift+p` or `cmd+shift+p`) and search for _GitLens: Open Settings_ and jump/scroll down to the _Interactive Rebase Editor_ section

![image](https://user-images.githubusercontent.com/641685/102309169-31ba2a00-3f36-11eb-8b26-050c7d83fa3f.png)

**Reply by ivancuric** [reply]
Currently impacted by https://github.com/microsoft/vscode/issues/105536 and I can't turn off gitlens' interactive rebase this way.

The settings do get updated, but it does nothing:
```js
  "workbench.editorAssociations": [
    {
      "viewType": "default",
      "filenamePattern": "git-rebase-todo"
    }
  ]
```

**Reply by eamodio** [reply]
Are you running a rebase from GitLens or the terminal? If inside GitLens then this is expected, but if from the terminal I'm not sure what might be going wrong.

**Reply by ivancuric** [reply]
Ah, was running from the right-click context menu in GitLens. Thought it'd disable it there as well.

**Reply by eamodio** [reply]
Not currently. If run from inside GitLens I "assume" you want the UI. Although, I should allow and opt-out for that as well.

**Reply by PiterGoCompany** [reply]
I'm using VS Code as my GIT default editor. After installing GitLens every rebase run from command line (eg. git rebase -i master) is causing that GitLens Interactive Editor is displayed. I can't disable Interactive Editor. Changing option in GitLens settings, and in VS setting doesn't help.

**Reply by eamodio** [reply]
Can you please add any more details you have to this issue: https://github.com/Axosoft/vscode-gitlens/issues/1687#issuecomment-974468565. Thanks!

**Reply by kellijohnson-NOAA** [reply]
This setting is reset each time VS Code updates. Is there any way to make it stick even across updates?

**Comment by devinrhode2** [comment]
for heavy rebasers, I don't think this is a net positive.

It is quite buggy wrt the state of the `git-rebase-todo` text file.

Also, you can't `ctrl+f` search commits and sha's.

It's great if you've literally never used interactive rebase, though!!!

I'd love ctrl+f and to see the bugs fixed

**Reply by eamodio** [reply]
Please open issues about the "buggy-ness" you are seeing - as I'm not aware of any issues.

As for Ctrl+F -- I thought I had enabled the search, but apparently not, so I'll get that in ASAP.

And of course if you are more comfortable using the text file, its just a single click away (one time or for all time)

**Reply by eamodio** [reply]
CTRL+F will be enabled in the next insiders builds

**Reply by devinrhode2** [reply]
I might be becoming an old hat, I guess I started with the text file, and now I'm coming back to it due to the bugginess.

I wonder if you might consider something akin to vscode's native git merge assistant? Where vscode basically just adds coloring across git conflict markers, and then inserts little text buttons for accepting a certain chunk of code?

It's going to be hard to re-produce, but somehow gitlens seems to be holding onto state of the file

Are you able to add logging (like with Sentry.io)? That's probably the best long term solution
